doc/nrf/gs_modifying.rst

```diff
@@ -181,12 +282,32 @@ Configuring build types
 
 .. build_types_overview_start
 
-Build types enable you to use different sets of configuration options for each board.
-You can create several build type :file:`.conf` files per board and select one of them when building the application.
-This means that you do not have to use one :file:`prj.conf` file for your project and modify it each time to fit your needs.
+The application is defined by a set of configuration files that provide definitions related to software and hardware, which together form the *configuration set* of the application.
+The configuration set specifies which features the application is going to use and how it is going to behave.
+For convenience, you can specify multiple configuration sets to quickly switch the application behavior.
+The most common example is to switch between ``release`` and ``debug`` variants of the application.
+
+In the |NCS|, you can define multiple build types.
+Select the configuration set for building the application by providing a reference to a specific build type file.
+This allows you to use different sets of configuration options for each board.
+
+The :file:`prj.conf` file is the main source for the :ref:`configure_application_sw`, owned by the application.
+You can create additional :file:`conf` files with specific software configuration set by following the name pattern :file:`prj_buildtype.conf`, where *buildtype* is replaced by the name of the build type (see :ref:`gs_modifying_build_types_creating`).
+Each provided :file:`prj_buildtype.conf` file is meant to be used independently, meaning that they do not overlay each other.
+Provided that you follow the file naming pattern, the build system will use the file suffix as the name of build type.
+For example, the :file:`prj_release.conf` file specifies the ``release`` build type.
+As the :file:`prj.conf` file does not have any suffix, it becomes the default build type.
+
+When building the application, you can define configuration file name using the ``CONF_FILE`` variable.
+The |NCS| build system propagates the information about the detected build type during the build process.
+This means that the software-related configuration is not the only one to be affected.
+The build system first looks for :file:`dts` or Partition Manager configuration files that match the selected build type.
```

Comment:
No it doesn't?

dts doesn't change depending on build type does it?

Reply:
> dts doesn't change depending on build type does it?

Correct, dts doesn't change with build type.
That [feature](https://github.com/zephyrproject-rtos/zephyr/pull/32436) was never merged as offline discussion considered this to create noise and build types wasn't considered needed for devicetree.

Reply:
These lines were edited to the following:

> When building the application, you can use the ``CONF_FILE`` variable to define the configuration file name.
> The |NCS| build system propagates the information about the detected build type during the build process.
> This is done for both the application and its child images.